chunkcache.cpp

```diff
@@ -127,5 +131,7 @@ void ChunkCache::routeStructure(QSharedPointer<GeneratedStructure> structure) {
 void ChunkCache::adaptCacheToWindow(int wx, int wy) {
   int chunks = ((wx + 15) >> 4) * ((wy + 15) >> 4);  // number of chunks visible
   chunks *= 1.10;  // add 10%
+
+  QMutexLocker guard(&mutex);
```

Comment:
setMaxCosts might also (if number decreases) erase elements from cache and thus needs to be protected